File: src/main/java/com/indiepost/validation/ContentTypeMultipartFileValidator.java
package com.indiepost.validation;

import org.springframework.web.multipart.MultipartFile;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

/**
 * Validator of content type. This is simple and not complete implementation
 * of content type validating. It's based just on <code>String</code> equalsIgnoreCase
 * method.
 *
 * @author Michal Kreuzman
 */
public class ContentTypeMultipartFileValidator implements ConstraintValidator<ContentType, MultipartFile> {

    private String[] acceptedContentTypes;

    private static boolean acceptContentType(String contentType, String[] acceptedContentTypes) {
        for (String accept : acceptedContentTypes) {
            if (contentType.equalsIgnoreCase(accept)) {
                return true;
            }
        }

        return false;
    }

    @Override
    public void initialize(ContentType constraintAnnotation) {
        this.acceptedContentTypes = constraintAnnotation.value();
    }

    @Override
    public boolean isValid(MultipartFile value, ConstraintValidatorContext context) {
        if (value == null || value.isEmpty())
            return true;

        return ContentTypeMultipartFileValidator.acceptContentType(value.getContentType(), acceptedContentTypes);
    }
}

File: src/main/kotlin/com/indiepost/validation/ContentTypeMultipartFileValidator.kt
package com.indiepost.validation

import org.springframework.web.multipart.MultipartFile

import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

internal class ContentTypeMultipartFileValidator : ConstraintValidator<ContentType, MultipartFile> {

    private lateinit var acceptedContentTypes: Array<String>

    private fun isAcceptedContentType(contentType: String?): Boolean {
        for (accept in acceptedContentTypes) {
            if (contentType!!.equals(accept, ignoreCase = true)) {
                return true
            }
        }
        return false
    }

    override fun initialize(constraintAnnotation: ContentType?) {
        this.acceptedContentTypes = constraintAnnotation!!.value
    }

    override fun isValid(value: MultipartFile?, context: ConstraintValidatorContext): Boolean {
        return if (value == null || value.isEmpty) false else isAcceptedContentType(value.contentType)
    }
}
